src/common/helpers/lambda-builder-helper.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { DockerImageFunctionProps } from 'aws-cdk-lib/aws-lambda/lib/image-function';
+import { Construct } from 'constructs';
+import { DockerLambdaCustomProps } from '../props/DockerLambdaCustomProps';
+
+export function buildDockerLambdaFunction(scope: Construct,
+  id: string, constructprops: DockerImageFunctionProps,
+  userprops: DockerLambdaCustomProps | undefined) {
+  if (userprops) { // user provided his own custom lambda function, merge his props with the construct ones
+    return new lambda.DockerImageFunction(
+      scope,
+      id,
+      {
+        // we force some values to be used from the construct
+        functionName: constructprops.functionName,
+        description: constructprops.description,
+        vpc: constructprops.vpc,
+        tracing: constructprops.tracing,
+        vpcSubnets: constructprops.vpcSubnets,
+        securityGroups: constructprops.securityGroups,
+        role: constructprops.role,
```

Comment:
we should add default memory size and timeout. The default value is from constructprops otherwise from userprops.

Reply:
done in [cf12de9](https://github.com/awslabs/generative-ai-cdk-constructs/pull/284/commits/cf12de914a19c3ef2d901380300dd6a8e6c75ba0)

Reply:
What would those defaults be?
```
      memorySize: lambdaMemorySizeLimiter(this, constructprops.memorySize),
      timeout: Duration.minutes(constructprops.timeout),
```

Reply:
@scottschreckengaust yes those values are implemented in each one of our constructs